src/main/java/com/redislabs/redisai/Dag.java

```diff
@@ -0,0 +1,77 @@
+package com.redislabs.redisai;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.ArrayUtils;
+import redis.clients.jedis.util.SafeEncoder;
+
+public class Dag implements DagRunCommands {
+  private final List<List<byte[]>> commands = new ArrayList<>();
+  private final List<Boolean> tensorgetflag = new ArrayList<>();
+
+  /** Direct acyclic graph of operations to run within RedisAI */
+  public Dag() {}
+
+  protected List<?> processDagReply(List<?> reply) {
+    List<Object> outputList = new ArrayList<>(reply.size());
+    for (int i = 0; i < reply.size(); i++) {
+      if (this.tensorgetflag.get(i) == true) {
+        outputList.add(Tensor.createTensorFromRespReply((List<?>) reply.get(i)));
+      } else {
+        outputList.add(reply.get(i));
+      }
+    }
+    return outputList;
+  }
+
+  @Override
+  public void setTensor(String key, Tensor tensor) {
```

Comment:
Can we change to chained calls here? like: `public Dag setTensor(String key, Tensor tensor) ` end with `return this;` The same goes for other methods. : )

Reply:
LvGTM :) changing